### ISSUE TITLE
Wii homebrew handler

### ISSUE THREAD

**Issue by Zowayix** [Zowayix/ROMniscience#30]
This is yet another issue I'm raising for myself where I had an idea about how to improve the code to be more flexible and extensible and this is a use case for such an idea.

So, if Wii homebrew games are going to be a thing... you're probably not going to get much out of just the .dol files. But they're all designed for the Homebrew Channel, so you have a meta.xml and an icon.png, which contain the things.

But how do we get at them, when the ROMFile passed to the handler only represents boot.dol? Well, because I like thinking too far ahead and planning the implementation before I even start on the concept, here's what I'm thinking:

Put a method in ROMFile: Stream getSiblingFile(string filename)
Returns a file with that name in the same folder as this file. Maybe validate against absolute paths or anything involving . or .. so we don't end up with any stupid security flaws (I'm not sure what the impact would be, given it'd be only called internally, but I just don't like the idea of blindly grabbing any relative path without making sure we're using things how we expect to be using them). Also, we're trying to get rid of InputStream and use StreamExtensions instead, so that cleanup might have to be done first.

But also if we're in an archive, get a file with the same name that's on the same level inside this archive. Remember that TODO comment where I wanted to refactor ROMFile so that it'd have a different class which is derived from some other base class for archived files and whatnot? That's probably a good idea to do first.

I'm not sure who archives their Wii homebrew folders since neither Homebrew Channel or Dolphin work that way, but it'd come in handy for other situations where we need another file that goes next to a given ROM.

**Comment by Zowayix** [comment]
Actually, is that StreamExtensions thing really going to work? InputStream does seem pointless but it does abstract away things like seeking support, or at least WrappedInputStream does. I might have to rethink how I'm going to do that cleanup, actually. Maybe I shouldn't do it how I thought.
